authentication/src/main/java/io/camunda/authentication/filters/TenantRequestAttributeFilter.java

```diff
@@ -56,24 +55,17 @@ protected void doFilterInternal(
     if (!multiTenancyCfg.isEnabled()) {
       return Set.of(TenantOwned.DEFAULT_TENANT_IDENTIFIER);
     }
-
-    return switch (principal) {
-      case final UsernamePasswordAuthenticationToken token -> {
-        if (!(token.getPrincipal() instanceof final CamundaUser user)) {
-          LOG.error("cannot find tenants: principal is not a camunda user");
-          yield null;
-        }
-        yield user.getTenants().stream().map(TenantDTO::tenantId).collect(Collectors.toSet());
-      }
-      case final OAuth2AuthenticationToken token -> {
-        LOG.error("cannot find tenants: tenant mapping isn't implemented for oidc");
-        yield null;
-      }
-      default -> {
-        LOG.error(
-            "cannot find tenants: unsupported principal type {}", principal.getClass().getName());
-        yield null;
-      }
-    };
+    if (!(principal instanceof final Authentication auth)) {
+      LOG.error(
+          "cannot find tenants: unsupported principal type: {}", principal.getClass().getName());
+      return null;
+    }
+    if (!(auth.getPrincipal() instanceof final CamundaPrincipal camundaPrincipal)) {
+      LOG.error("cannot find tenants: unsupported principal type: {}", auth.getClass().getName());
+      return null;
+    }
```

Comment:
💭 I wonder if here we can instead extract this to a single helper method and have one if guard?

Reply:
The error message is different because we're checking the type of different objects. A helper method would probably have to have the same structure in its body.

Reply:
True, we don't really gain much but it would just lift it and reduce this overall methods size